File: Modules/SMES/MES/SIE.Web.MES/WorkOrders/Commands/WorkOrderCompletionCommand.js
SIE.defineCommand('SIE.Web.MES.WorkOrders.Commands.WorkOrderCompletionCommand', {
    //extend: 'SIE.cmd.Save',
    meta: { text: "手动完工", group: "edit", iconCls: "icon-Upload icon-blue" },
    canExecute: function (listView) {

        var selectModels = listView.getCurrent();
        if (selectModels == null) return false;
        else {
            if (selectModels.getFinishQty() >= selectModels.getPlanQty()) {
                return true;
            }
            else
                return false;
        }
        return true;
    },
    execute: function (view) {
        view.execute({
            data: view.getCurrent().getData(),
            success: function (res) {
                debugger;
                if (res.Success)
                    alert("该工单已经完工！");
            }
        });
    }
});

File: Modules/SMES/MES/SIE.Web.MES/WorkOrders/Commands/WorkOrderCompletionCommand.cs
using SIE.Core.WorkOrders;
using SIE.Domain;
using SIE.MES.WorkOrders;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.ServiceModel.Channels;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.WorkOrders.Commands
{
    /// <summary>
    /// 手动完工
    /// </summary>
    public class WorkOrderCompletionCommand : ViewCommand
    {
        protected override object Excute(ViewArgs args, string scope)
        {
            var entity = args.Data.ToJsonObject<SIE.MES.WorkOrders.WorkOrder>();
            entity.State = WorkOrderState.Finish;
            RF.Save(entity);
         
            return true;
        }
    }
}
